rpc-tests/tests/test-rpc.js

```diff
@@ -21,69 +26,50 @@ describeWithNetwork(network, `${network} RPC`, function(context) {
+        expect(finalised).to.be.true;
+        expect(dappInfoOpt.isSome).to.be.true;
 
-        console.log('Transfer sent with hash', hash.toHex());
+        const dappInfo = dappInfoOpt.unwrap();
 
-        expect(hash).to.exist;
+        expect(dappInfo.developer.toString()).to.equals(ALICE);
+        expect(dappInfo.state.toString()).to.equals('Registered');
     });
 
-	it('should be able to transfer tokens from alice to bob', async function () {
-        const api = context.api;
+	it('should be able to transfer tokens from alice to charlie', async () => {
+        const originalBalance = await context.api.query.system.account(CHARLIE);
+        const finalised = await sendTransaction(
+            context.api.tx.balances.transfer({ Id: CHARLIE }, 100),
+            context.alice
+        );
+        const newBalance = await context.api.query.system.account(CHARLIE);
 
-        // Construct the keyring after the API (crypto has an async init)
-        const keyring = new Keyring({ type: 'sr25519' });
-      
-        // Add Alice to our keyring with a hard-derivation path (empty phrase, so uses dev)
-        const alice = keyring.addFromUri('//Alice');
-
-        // Create a extrinsic, transferring 100 units to Bob
-        const transfer = api.tx.balances.transfer(BOB, 100);
-      
-        // Sign and send the transaction using our account
-        const hash = await transfer.signAndSend(alice, { nonce: -1 });
-
-        console.log('Transfer sent with hash', hash.toHex());
-
-        expect(hash).to.exist;
+        expect(finalised).to.be.true;
+        expect(newBalance.data.free.sub(originalBalance.data.free).toNumber()).to.equal(100);
     });
 
-    it('should be able to transfer tokens from bob to alice', async function () {
-        const api = context.api;
-
-        // Construct the keyring after the API (crypto has an async init)
-        const keyring = new Keyring({ type: 'sr25519' });
-
-        // Add Alice to our keyring with a hard-derivation path (empty phrase, so uses dev)
-        const bob = keyring.addFromUri('//Bob');
-
-        // Create a extrinsic, transferring 100 units to Bob
-        const transfer = api.tx.balances.transfer(ALICE, 200);
-
-        // Sign and send the transaction using our account
-        const hash = await transfer.signAndSend(bob, { nonce: -1 });
-
-        console.log('Transfer sent with hash', hash.toHex());
+    it('should be able to transfer tokens from bob to dave', async () => {
+        const originalBalance = await context.api.query.system.account(DAVE);
+        const finalised = await sendTransaction(
+            context.api.tx.balances.transfer({ Id: DAVE }, 200),
+            context.bob
+        );
+        const newBalance = await context.api.query.system.account(DAVE);
 
-        expect(hash).to.exist;
+        expect(finalised).to.be.true;
+        expect(newBalance.data.free.sub(originalBalance.data.free).toNumber()).to.equal(200);
```

Comment:
Just a small comment here. When converting BN to number you can loose precision. This function is not affected because amount is small, but in general it is good practice to avoid converting BN to number. The same applies to the function above.

Reply:
Yes i have converted it to number as amount was small. chai assertion needs native types for comparison.